x/observer/keeper/vote_outbound.go

```diff
@@ -0,0 +1,52 @@
+package keeper
+
+import (
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	"github.com/zeta-chain/zetacore/common"
+	observertypes "github.com/zeta-chain/zetacore/x/observer/types"
+)
+
+// VoteOnOutboundBallot casts a vote on an outbound transaction observed on a connected chain (after
+// it has been broadcasted to and finalized on a connected chain). If this is
+// the first vote, a new ballot is created. When a threshold of votes is
+// reached, the ballot is finalized.
+// returns if the vote is finalized, if the ballot is new, the ballot status and the name of the observation chain
+func (k Keeper) VoteOnOutboundBallot(
+	ctx sdk.Context,
+	ballotIndex string,
+	outTxChainID int64,
+	receiveStatus common.ReceiveStatus,
+	voter string,
+) (isFinalized bool, isNew bool, ballot observertypes.Ballot, observationChainName string, err error) {
+	// Observer Chain already checked then inbound is created
+	/* EDGE CASE : Params updated in during the finalization process
+	   i.e Inbound has been finalized but outbound is still pending
+	*/
+	observationChain := k.GetSupportedChainFromChainID(ctx, outTxChainID)
+	if observationChain == nil {
+		return false, false, ballot, "", observertypes.ErrSupportedChains
+	}
+	if observertypes.CheckReceiveStatus(receiveStatus) != nil {
+		return false, false, ballot, "", observertypes.ErrInvalidStatus
+	}
+
+	// check if voter is authorized
+	if ok := k.IsAuthorized(ctx, voter); !ok {
+		return false, false, ballot, "", observertypes.ErrNotObserver
+	}
+
+	// fetch or create ballot
+	ballot, isNew, err = k.FindBallot(ctx, ballotIndex, observationChain, observertypes.ObservationType_OutBoundTx)
```

Comment:
Since we are already refactoring some of the ballot logic, it would be good to add the `voter list` and `threshold` as part of the parameter list so that the TSS voter can also use this function.

This can, however, be done as a separate PR, too, if you want to limit the changes on this one.

Reply:
I might think this is a bit related to the first comment? https://github.com/zeta-chain/node/issues/1806